tracker-joystick/ftnoir_tracker_joystick.cpp

```diff
@@ -46,13 +65,36 @@ void joystick::data(double *data)
 
     const QString guid = s.guid;
     int axes[8];
+#ifdef _WIN32
     const bool ret = joy_ctx.poll_axis(guid, axes);
+#elif defined __linux__
+    struct js_event event;
+    bool ret = true;
+    if (read(joy_fd, &event, sizeof(event)) > 0)
+    {
+        switch (event.type)
+        {
+        case JS_EVENT_AXIS:
+            if (event.number >= 8) break;
+            axes_state[event.number] = event.value;
+
+            break;
+        default:
+            /* Ignore init/button events. */
+            break;
+        }
+    }
 
+    for (int i = 0; i < 6; i++)
+    {
+        axes[i] = axes_state[i];
+    }
+#endif
     if (ret)
     {
         for (int i = 0; i < 6; i++)
         {
-            int k = map[i] - 1;
```

Comment:
I think i might have fixed a bug here, 

at the top of this method the joysticks axes are defined as 

```
    int map[6] = {
        s.joy_1 - 1,
        s.joy_2 - 1,
        s.joy_3 - 1,
        s.joy_4 - 1,
        s.joy_5 - 1,
        s.joy_6 - 1,
    };
```
that combined with the `int k = map[i] -1` means we are effectively subtracting 2

So the dropdown looked like this once it's values were evaluated in this loop
* Disabled == -2
* Axis 1 == -1
* Axis 2 == 0
...